### DISCUSSION TITLE
Request bundling

### DISCUSSION THREAD

**Discussion by wizzymore** [livewire/livewire#6143] [1. Help] [answered]
I am a bit worried about request bundling and i want to ask some questions:

1. Can it be disabled?
2. Can it be disabled only for some specific components?

The main worry i have about the request bundling is that some components render or have less work to do then other components and bundling them means the "low work" components will be updated only when the slowest link in the components will be done.

I get the benefits, having less handshakes and doing only one round-trip to the browser, but we can't use the concurrency that http2 or http3 gives us.

As an example i have the following components:

```php
<?php

namespace App\Livewire;

use Livewire\Component;

class TestOne extends Component
{
    public int $count = 0;

    protected $listeners = ['increaseCount' => 'increaseCount'];

    public function increaseCount()
    {
        sleep(5);
        $this->count++;
    }

    public function render()
    {
        return <<<'HTML'
        <div>
            <h1>Test One: {{ $count }}</h1>
        </div>
        HTML;
    }
}
```

```php
<?php

namespace App\Livewire;

use Livewire\Component;

class TestTwo extends Component
{
    public int $count = 0;

    protected $listeners = ['increaseCount' => 'increaseCount'];

    public function increaseCount()
    {
        $this->count++;
    }

    public function render()
    {
        return <<<'HTML'
        <div>
            <h1>Test Two: {{ $count }}</h1>
        </div>
        HTML;
    }
}
```

If i dispatch the event `increaseCount` for both components to update it will take minimum 5 seconds instead of one taking ~100ms and another one ~5.1s

**Comment by wizzymore** [comment]
Bump

**Comment by joshhanley** [comment]
@wizzymore at the moment, no it can't be disabled. Any chance you could submit a failing test PR that demonstrates the issue so we can get Caleb to have a look to see if he wants to implement a way to split requests? Thanks!

**Comment by wizzymore** [comment]
Is this still the case? can we disable bundling now?

I don't get why we need bundling in the world of HTTP/2

**Comment by AyoobMH** [comment]
Is it possible to prioritize this? There are use cases where many livewire components are causing slow loading. Appreciate your time!

**Comment by eilerth** [comment]
I was facing this same issue while trying to upgrade my existing project to livewire v3.
I've submitted this PR as a potential solution to the problem: https://github.com/livewire/livewire/pull/7102

Essentially, it allows you to add "isolate" to a livewire component to prevent it from bundling it's requests with other livewire components.
`<livewire:my-component isolate />`

**Comment by alxndrmlr** [comment]
I just ran into this as well...

I have a `FiltersComponent` that dispatches a `filters.updated` event that I then have two other components listening to it; `UserListComponent` and a `MapComponent`

The problem is that the `MapComponent` is a going to be a slower request whereas the `UserListComponent` is fast. Since they get bundled, the app feels less responsive.

I solved this by just changing my code to only have the `UserListComponent` listen to the `filters.updated` event, then it dispatches a new event `directory.usersUpdated` and the  `MapComponent` listens for that instead.

This makes the updates go in series ie: 

```
`FiltersComponent`  > `UserListComponent` > `MapComponent`   
```

but in an ideal world it would be in parallel ie;

```
                    `UserListComponent`
                  / 
`FiltersComponent` 
                  \
                    `MapComponent`   
                                               
```                                               

I recorded a Loom video https://www.loom.com/share/22e87bb062ae42d884ec729ed874c20d to show what I mean. I can't really "speed up" the `MapComponent`  to solve this as it needs to do aggregation and summation over large datasets.

Trying to put my @calebporzio hat on (ie: finding something elegant), what I think I would want to reach for is something like this... 

`#[On('directory.filtersReset', bundleRequest: false)]`

ie: a way to define at the event listener level, that it wants to be handled independently ... 

```
<?php

namespace App\Livewire\Directory;

use App\Services\DirectoryService;
use Illuminate\Support\Facades\DB;
use Livewire\Attributes\On;
use Livewire\Component;

class DirectoryMapComponent extends Component
{
    public $shouldSuggestFilterMapBounds = false;

    #[On('directory.filtersReset', bundleRequest: false)]
    public function handleFiltersReset()
    {
        $this->computeFilterOrganizationCounts();
    }

    #[On('directory.filtersUpdated', bundleRequest: false)]
    public function handleFilterUpdated($filters)
    {
        $this->shouldSuggestFilterMapBounds = $filters['limitToMap'];
        $this->computeFilterOrganizationCounts($filters);
    }

// ... rest of the file
}
```

**Reply by wizzymore** [reply]
Yeah but that is not really a solution, it creates a waterfall instead of doing them in parallel.

Also I love the idea you have, it looks nice.

Edit: Had to edit as my last version of the comment made no sense xD

**Comment by wizzymore** [comment, marked as answer]
Hello everybody,

This was solved, i don't know when or where but now we have this:
https://livewire.laravel.com/docs/bundling

Thank you to everyone that made this happen!
